frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx

```diff
@@ -0,0 +1,190 @@
+  const [helmChartEntries, setHelmChartEntries] = React.useState<HelmChartMetaData[]>([]);
+  const helmAction: HelmActionType =
+    chartURL !== 'null' ? HelmActionType.Install : HelmActionType.Upgrade;
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchHelmReleases = async () => {
+      let res;
+      try {
+        res =
+          helmAction === HelmActionType.Install
+            ? await coFetchJSON(`/api/helm/chart?url=${chartURL}`)
+            : await coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+      const data: HelmChart = !_.isEmpty(res.chart) ? res.chart : res;
+      setYAMLData(!_.isEmpty(res.config) ? _.merge(data.values, res.config) : data.values);
+      if (helmAction === HelmActionType.Upgrade) {
+        setChartName(data.metadata.name);
+        setActiveChartVersion(data.metadata.version);
+      }
+      setChartDataLoaded(true);
+      !_.isEmpty(data.values) && setChartHasValues(true);
+    };
+
+    fetchHelmReleases();
+
+    return () => {
+      ignore = true;
+    };
+  }, [chartURL, namespace, releaseName, helmAction]);
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchChartVersions = async () => {
+      let json: HelmRelease;
+
+      try {
+        const response = await coFetch('/api/helm/charts/index.yaml');
+        const yaml = await response.text();
+        json = safeLoad(yaml);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+      setHelmChartEntries(_.get(json, ['entries', chartName]));
+    };
+    helmAction !== HelmActionType.Install && fetchChartVersions();
+    return () => {
+      ignore = true;
+    };
+  }, [chartName, helmAction]);
+
+  const initialValues: HelmInstallUpgradeFormData = {
+    helmReleaseName: releaseName || helmChartName || '',
+    chartValuesYAML: !_.isEmpty(YAMLData) ? safeDump(YAMLData) : undefined,
+    chartVersion: activeChartVersion,
+  };
+
+  const handleSubmit = (values, actions) => {
+    const { helmReleaseName, chartVersion, chartValuesYAML }: HelmInstallUpgradeFormData = values;
+    let valuesObj;
+
+    if (chartValuesYAML) {
+      try {
+        valuesObj = safeLoad(chartValuesYAML);
+      } catch (err) {
+        actions.setStatus({ submitError: `Invalid YAML - ${err}` });
+        return;
+      }
+    }
+    let newChartURL = '';
+    newChartURL = chartVersion && getChartURL(helmChartEntries, chartVersion);
+
+    const payload = {
+      namespace,
+      name: helmReleaseName,
+      ...(helmAction === HelmActionType.Install
+        ? // eslint-disable-next-line @typescript-eslint/camelcase
+          { chart_url: chartURL }
+        : // eslint-disable-next-line @typescript-eslint/camelcase
+          { chart_url: newChartURL }),
+      ...(valuesObj ? { values: valuesObj } : {}),
+    };
+
+    coFetchSendJSON(
```

Comment:
I think you can use `coFetchJSON` and send the method similar to this as well. That way you won't need to have change in console code. This function just adds some extra headers, I doubt if those headers are absolutely necessary.

Reply:
You can get the `cofetchJSON.post` and `coFetchJSON.put` by doing `_.get(coFetchJSON, 'post')`. This way you can avoid using `coFetchSendJSON` directly in your code and still have the logic to use correct method based on the helm action.